### PR TITLE
Reclusts matrixplot

### LINKED ISSUE #858: haddock3-re clustfcc not generating plots
**Is your feature request related to a problem? Please describe.**
It is unfortunate that the current implementation of `haddock3-re clustfcc` is not allowing to generate a plot of the matrix.

**Describe the solution you'd like**
Add it.

### PR DESCRIPTION
**Pull request by VGPReys**
You are about to submit a new Pull Request. Before continuing make sure you read the [contributing guidelines](CONTRIBUTING.md) and that you comply with the following criteria:

- [X] You have sticked to Python. Please talk to us before adding other programming languages to HADDOCK3
- [ ] Your PR is about CNS
- [ ] Your code is well documented: proper docstrings and explanatory comments for those tricky parts
- [ ] You structured the code into small functions as much as possible. You can use classes if there is a (state) purpose
- [X] Your code follows our coding style
- [X] You wrote tests for the new code
- [X] `tox` tests pass. *Run `tox` command inside the repository folder*
- [X] `-test.cfg` examples execute without errors. *Inside `examples/` run `python run_tests.py -b`*
- [X] PR does not add any dependencies, unless permission granted by the HADDOCK team
- [X] PR does not break licensing
- [ ] Your PR is about writing documentation for already existing code :fire:
- [ ] Your PR is about writing tests for already existing code :godmode:

---

Closes #858  by adding the `-p`/`--plot_matrix` flag optional parameter in `haddock3-re` for both `clustrmsd` & `clustfcc`, allowing to generate plots as it would be in standard modules.

### CODE REVIEW

File: src/haddock/clis/re/clustfcc.py [hunk 1]
```diff
@@ -56,6 +58,15 @@ def add_clustfcc_arguments(clustfcc_subcommand):
         type=int,
         )
     
+    clustfcc_subcommand.add_argument(
+        "-p",
+        "--plot_matrix",
+        help="Generate the a plot of the clusters.",
```
Comment (mgiulini) [inline src/haddock/clis/re/clustfcc.py:64]:
```suggestion
        help="Generate the matrix plot with the clusters.",
```

File: src/haddock/clis/re/clustfcc.py [hunk 2]
```diff
@@ -167,5 +183,36 @@ def reclustfcc(
         if caprieval_folder:
             log.info("Rewriting capri tables")
             rewrite_capri_tables(caprieval_folder, clt_dic, outdir)
+    
+    else:
+        output_models = models
```
Comment (mgiulini) [inline src/haddock/clis/re/clustfcc.py:188]: why this? are the `output_models` necessary in an interactive running?

File: src/haddock/clis/re/clustrmsd.py [hunk 3]
```diff
@@ -54,6 +57,15 @@ def add_clustrmsd_arguments(clustrmsd_subcommand):
         required=False,
         type=int,
         )
+    
+    clustrmsd_subcommand.add_argument(
+        "-p",
+        "--plot_matrix",
+        help="Generate the a plot of the clusters.",
```
Comment (mgiulini) [inline src/haddock/clis/re/clustrmsd.py:64]:
```suggestion
        help="Generate the matrix plot with the clusters.",
```

**Review by mgiulini — APPROVED** [review]

File: src/haddock/clis/re/clustfcc.py (hunk 2 above)
Comment (VGPReys) [reply src/haddock/clis/re/clustfcc.py:188]: Mandatory, as the matrix plot is using `output_models` to get the current list of models

**Pull request merged**